flaskbb/forum/views.py

```diff
@@ -90,7 +92,14 @@ def get(self, category_id, slug=None):
 
 
 class ViewForum(MethodView):
-    decorators = [allows.requires(CanAccessForum())]
+    decorators = [allows.requires(
+        CanAccessForum(),
+        on_fail=FlashAndRedirect(
+            message=_("You are not allowed to access that forum"),
+            level="warning",
+            endpoint=lambda *a, **k: current_category.url
+        )
+    )]
```

Comment:
Allows added `on_fail` in 0.4.0 but I don't think we ever took advantage of it. When used as a decorator, if the return value of `on_fail` isn't None, it's treated as the response to the routing request it also receives whatever args were passed to the decorated function. FlashAndRedirect is exactly what it says on the tin, it flashes a message and then returns a redirect and the endpoint argument can either be a string or a function that returns a string and accepts the arguments that were passed to the route handler.